packages/ripple-binary-codec/src/types/st-object.ts

```diff
@@ -101,8 +101,8 @@ class STObject extends SerializedType {
 
     const xAddressDecoded = Object.entries(value).reduce((acc, [key, val]) => {
       let handled: JsonObject | undefined = undefined;
-      if (isValidXAddress(val)) {
-        handled = handleXAddress(key, val);
+      if (val && isValidXAddress(val.toString())) {
+        handled = handleXAddress(key, val.toString());
```

Comment:
typescript was updated in this change to a common, and more recent version. these changes are necessary to pass `tsc`.